### ISSUE TITLE
Develop a new helpdesk module as a replacement of the OCA helpdesk module

### ISSUE THREAD

**Issue by gonzalesedwin1123** [openspp-project/openspp-registry#357]
The OCA helpdesk module, which was used in the Odoo 15 version of OpenSPP as the base of the Grievance Redress Mechanism module was decided not be used in the Odoo 17 version of OpenSPP because it is licensed under AGPL. A new Grievance Redress Mechanism module for the Odoo17 version of OpenSPP must be developed.

Mode of Input:

- [x] Import from files (Excel, CSV)
- [x] APIs integration with different applications / modules using XMLRPC and JSONRPC
- [x] Email
- [x] Phone Call / Call center
- [x] Web backend
- [x] Self Service using the Web Portal UI

Record Access Control:

- [ ] User roles assigned to implement the changes can only access the registrant records specified in the tickets.

Integration with Change Request Module:

- [ ] Based on the ticket category, a change request must be generated and linked to the ticket.

**Comment by anthonymarkQA** [comment]
Findings: Returned to DEV

Issues found:

- missing GRM manager role 
- missing GRM user access role setup
- no Kanban board found as per steps in the ticket indicated

Steps to replicate:

1. create a runboat instance with this PR
2. login to the instance as admin
3. Install modules for spmis
4. Install the module spp_grm.
5. Create a group and individual in registry
6. Open an individual or group.
7. Click the Tickets button to display the existing tickets for the registrant or create a new one.
8. To manage the tickets, a Grievance Redress Mechanism (GRM) manager can assign the tickets to GRM users who will perform the necessary actions based on the description of the ticket. This is done by clicking the Helpdesk menu. - no GRM manager or GRM user role found or configured.
9. In the Helpdesk UI, a list of tickets will be displayed. Open a ticket to assign to a GRM user.
10. GRM users can only see the tickets assigned to them. A kanban view is available and can be used to easily change the stage of the tickets. -- no Kanban board is visible in the UI
11. Communications, schedule activities, and upload or open files can be done using the chatter.

![Image](https://github.com/user-attachments/assets/32f57ee7-4deb-40aa-8acb-e4cf8864d2e7)

**Comment by anthonymarkQA** [comment]
Findings: In progress, needs further discussion for expected results


**local support role** 
1. View individual - ok
2. View group - Access error, You are not allowed to access 'ID Queue' (spp.print.queue.id) records.
3. create individual - ok
4. create group - ok
5. Edit individual - ok
6. Edit group - Blocked by View group
7. Create GRM ticket from individual view -fail, (Sorry, g***********@mailinator.com (id=7) doesn't have 'create' access to:
Grievance Redress Mechanism Ticket (spp.grm.ticket))
8. Create GRM ticket from group view - fail,  (Sorry, g***********@mailinator.com (id=7) doesn't have 'create' access to:
Grievance Redress Mechanism Ticket (spp.grm.ticket))
9. Create GRM ticket from sidenav Helpdesk -fail.  (Sorry, g***********@mailinator.com (id=7) doesn't have 'create' access to:
Grievance Redress Mechanism Ticket (spp.grm.ticket))
10. Import group -ok
11. Import individual -ok
12. Registry>Configurations(registrant tags, gender, etc..) - view  access only.


**Global support role**
1. View individual - ok
2. View group - Access error, You are not allowed to access 'ID Queue' (spp.print.queue.id) records.
3. create individual - ok
4. create group - ok
5. Edit individual - ok
6. Edit group - Blocked by View group
7. Create ticket from individual view - fail, (Sorry, g***********@mailinator.com (id=7) doesn't have 'create' access to:
Grievance Redress Mechanism Ticket (spp.grm.ticket))
8. Create ticket from group view - fail.  (Sorry, g***********@mailinator.com (id=7) doesn't have 'create' access to:
Grievance Redress Mechanism Ticket (spp.grm.ticket))
9. Create GRM ticket from sidenav Helpdesk -fail.  (Sorry, g***********@mailinator.com (id=7) doesn't have 'create' access to:
Grievance Redress Mechanism Ticket (spp.grm.ticket))
10. Import group -ok
11. Import individual -ok
12. Registry>Configurations(registrant tags, gender, etc..) - view  access only.

**Portal Users from Self Service module feature:**

1. can only view tickets related particularly to them - ok
2. can create tickets against themselves. - ok
3. can edit their own registrant data - ok
4. can view invoices against their own registrant data. - fail , i cannot see cash entitlements i received for the registrant
5. can change password to own account - ok
6. can delete own account - ok

**Comment by anthonymarkQA** [comment]
Here are some behaviors that were defined together with @singhkaranvir25 

Global support role -

- Should be able to view Group and Individual and is read-only access, no import, edit or create allowed.
- Should be able to access Change request module and create change requests
- Global support must be able to view all support tickets created

Local support role - 

- Should be able to view Group and Individual and is read-only access, no import, edit or create allowed.
- Should be able to access Change request module and create change requests
- Local support must be able to view support tickets created under the same area, also add additional filter to display tickets specifically assigned to this local support.

Self Service webpage-

- Add a ticket number column in support tickets table for the purpose of being used as reference number when calling helpdesk support
- Name column can be removed as all tickets displayed are all owned by this individual/group only
- Self Service webpage for now should be read only. should not be able to edit any data except changing password.
- If a portal user's account is deleted, all support tickets associated with it shall be marked invalid.
- If a portal user's account is delete, the registry data should not be deleted and must remain intact. only the portal account shall be deleted.

Import feature for support tickets in helpdesk
-import to create new ticket
--Should contain the following headers:
partner_id | name | description | category_id | channel_id
--Required headers must be also obtainable by doing export of an existing ticket.
![image](https://github.com/user-attachments/assets/8f98a133-aed9-4402-9216-5c54ff59a2cb)
--channel id must contain the value 'other'
--value for 'partner_id' can be either group's name or individual's name that exists in openspp registry

-Import to update new ticket
--Should contain the following headers:
id | partner_id | name | description | category_id | channel_id
--Required headers must be also obtainable by doing export of an existing ticket.
![image](https://github.com/user-attachments/assets/9b16aecc-63d3-4c13-bb3f-06303eee3699)
--channel id must contain the value 'other'
--value for 'partner_id' can be either group's name or individual's name that exists in openspp registry

**Comment by anthonymarkQA** [comment]
Behaviors to fix: @gonzalesedwin1123 

### local support role:
- cannot view group but should be able to.
- cannot create ticket but should be able to. (should it also allow edit and deletion of tickets?) 
- remove access rights for edit, create, delete and import access for both group and individual
- should be able to access and create change requests
- should be able to see all tickets belonging to the same area, additionally add a filter that allows to display all tickets assigned to this local support role.

### Global support role:
- cannot view group but should be able to.
- cannot create ticket but should be able to. (should it also allow edit and deletion of tickets?)
- remove access rights for edit, create, delete and import access for both group and individual
- should be able to access and create change requests
- should be able to see all tickets regardless of any area,  additionally add a filter that allows to display all tickets assigned to this Global support role.

### Self service webpage
- Add a ticket number column in support tickets table for the purpose of being used as reference number when calling helpdesk support. Currently has no ticket number to use as reference.
- Remove name column as all tickets displayed are all owned by this individual/group only. Currently displays the name redundantly.
- Self Service webpage for now should be read only. should not be able to edit any data(address, birthdate, name etc..) except **changing password**. Currently i am able to edit the registrants data.
- If a portal user's account is deleted, all support tickets associated with it shall be marked invalid. Currently the ticket still exists.
- If a portal user's account is delete, the registry data associated should not be deleted and must remain intact. only the portal account shall be deleted. Currently it also deletes the data associated in the registry.

### Import support ticket as new or to edit existing
all good. just needs documentation to know what are the required headers for either new or edit existing tickets

### email
will be taken care on ticket 665

**Comment by gonzalesedwin1123** [comment]
@celinenilla  Considering that the findings of the QA is more on the user roles and access rights, I suggest that we merge the PR now and that new tickets be created to continue working on it.

**Comment by anthonymarkQA** [comment]
following tickets have been created in order to correct the identified issues:

- https://github.com/OpenSPP/openspp-modules/issues/738
- https://github.com/OpenSPP/openspp-modules/issues/739
- https://github.com/OpenSPP/openspp-modules/issues/740

this ticket is ready for release and approved by qa
